pkg/server/middleware/webhook.go

```diff
@@ -0,0 +1,46 @@
+package middleware
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+
+	"github.com/tbd54566975/ssi-service/pkg/server/framework"
+	svcframework "github.com/tbd54566975/ssi-service/pkg/service/framework"
+	"github.com/tbd54566975/ssi-service/pkg/service/webhook"
+)
+
+// Define a custom response writer that wraps the original response writer and writes to a buffer
+type responseWriter struct {
+	http.ResponseWriter
+	buf *bytes.Buffer
```

Comment:
I don't think it would add anything, we can wrap the response body with a new field if needed. I worry if we include the headers we could open ourselves up to security risk like auth tokens in the headers or something